### PR TITLE
refactor: resolve clippy warnings in rs-drive

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented
Clippy warnings in rs-drive

## What was done?
Repo wide clippy warnings 326 -> 316
in package rs-drive before 238 -> 225

## How Has This Been Tested?
`cargo t` in base of repo

## Breaking Changes
None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [x] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/rs-drive/src/drive/verify/mod.rs [hunk 1]
```diff
@@ -15,3 +15,5 @@ pub mod state_transition;
 
 /// Represents the root hash of the grovedb tree
 pub type RootHash = [u8; 32];
+/// Simply a collection of 32 Bytes. It's just that simple
+pub type Bytes32 = [u8; 32];
```
Comment (QuantumExplorer) [inline packages/rs-drive/src/drive/verify/mod.rs:19]: We already have a Bytes32... we should not have two of them.

Comment (PastaPastaPasta) [reply packages/rs-drive/src/drive/verify/mod.rs]: where? I saw a different one but it was in rs-dpp I think; and imo it doesn't make sense to depend on each other for something like Bytes32?

File: packages/rs-drive/src/drive/batch/drive_op_batch/mod.rs [hunk 2]
```diff
@@ -53,7 +53,9 @@ pub trait DriveLowLevelOperationConverter {
 /// The drive operation context keeps track of changes that might affect other operations
 /// Notably Identity balance changes are kept track of
 pub struct DriveOperationContext {
-    //todo: why is this not being used?
+    #[allow(dead_code)]
+    #[deprecated(note = "This function is marked as unused.")]
+    #[allow(deprecated)]
```
Comment (shumkov) [inline packages/rs-drive/src/drive/batch/drive_op_batch/mod.rs:58]: Why we allow deprecated code here?

File: packages/rs-drive/src/drive/contract/mod.rs [hunk 3]
```diff
@@ -95,6 +95,9 @@ mod tests {
     use crate::tests::helpers::setup::setup_drive_with_initial_state_structure;
     use dpp::version::PlatformVersion;
 
+    #[allow(dead_code)]
```
Comment (shumkov) [inline packages/rs-drive/src/drive/contract/mod.rs:98]: Shall we just remove dead code?

File: packages/rs-drive/src/drive/grove_operations/mod.rs [hunk 4]
```diff
@@ -235,15 +235,15 @@ impl BatchInsertTreeApplyType {
     /// - A variant of `DirectQueryType::StatelessDirectQuery` if the current type is `BatchInsertTreeApplyType::StatelessBatchInsertTree`.
     /// - `DirectQueryType::StatefulDirectQuery` if the current type is `BatchInsertTreeApplyType::StatefulBatchInsertTree`.
     /// ```
-    pub(crate) fn to_direct_query_type(&self) -> DirectQueryType {
+    pub(crate) fn to_direct_query_type(self) -> DirectQueryType {
```
Comment (shumkov) [inline packages/rs-drive/src/drive/grove_operations/mod.rs:238]: We we consume self here it should be called `into_direct_query_type`

**Review by shumkov — CHANGES_REQUESTED** [review]

File: packages/rs-drive/src/drive/contract/mod.rs (hunk 3 above)
Comment (PastaPastaPasta) [reply packages/rs-drive/src/drive/contract/mod.rs:98]: good question; I can (and would probably prefer to) but this feels less destructive / likely to be objected to by sam

File: packages/rs-drive/src/drive/grove_operations/mod.rs (hunk 4 above)
Comment (PastaPastaPasta) [reply packages/rs-drive/src/drive/grove_operations/mod.rs:238]: self implements copy; so there's no need to take as ref

File: packages/rs-drive/src/drive/verify/mod.rs (hunk 1 above)
Comment (shumkov) [reply packages/rs-drive/src/drive/verify/mod.rs]: They already depend on each other

File: packages/rs-drive/src/drive/contract/mod.rs (hunk 3 above)
Comment (shumkov) [reply packages/rs-drive/src/drive/contract/mod.rs:98]: Yeah, but doing this we just shut linter which is not actually what you suppose to do

File: packages/rs-drive/src/drive/grove_operations/mod.rs (hunk 4 above)
Comment (shumkov) [reply packages/rs-drive/src/drive/grove_operations/mod.rs:238]: @QuantumExplorer what do you think?

File: packages/rs-drive/src/drive/contract/mod.rs (hunk 3 above)
Comment (QuantumExplorer) [reply packages/rs-drive/src/drive/contract/mod.rs:98]: I agree with Pasta.

File: packages/rs-drive/src/drive/grove_operations/mod.rs (hunk 4 above)
Comment (QuantumExplorer) [reply packages/rs-drive/src/drive/grove_operations/mod.rs:238]: Pasta is right according to linter.

File: packages/rs-drive/src/drive/contract/mod.rs (hunk 3 above)
Comment (PastaPastaPasta) [reply packages/rs-drive/src/drive/contract/mod.rs:98]: > shut linter which is not actually what you suppose to do

I do agree with this; but goal ATM is to get to a point where we can turn on CI enforcement on warnings as errors; as such need to suppress in cases like this

File: packages/rs-drive/src/drive/verify/mod.rs (hunk 1 above)
Comment (QuantumExplorer) [inline packages/rs-drive/src/drive/verify/mod.rs:19]: No, don't do this.

File: packages/rs-drive/src/drive/verify/contract/verify_contract/v0/mod.rs [hunk 5]
```diff
@@ -168,9 +168,9 @@ impl Drive {
     pub fn verify_contracts(
         proof: &[u8],
         _is_proof_subset: bool, //this will be used later
-        contract_ids: &[[u8; 32]],
+        contract_ids: &[Bytes32],
         platform_version: &PlatformVersion,
-    ) -> Result<(RootHash, BTreeMap<[u8; 32], Option<DataContract>>), Error> {
+    ) -> Result<(RootHash, BTreeMap<Bytes32, Option<DataContract>>), Error> {
```
Comment (QuantumExplorer) [inline packages/rs-drive/src/drive/verify/contract/verify_contract/v0/mod.rs:173]: Keep  [u8; 32]

File: packages/rs-drive/src/drive/verify/mod.rs (hunk 1 above)
Comment (PastaPastaPasta) [reply packages/rs-drive/src/drive/verify/mod.rs]: but then there's also bytes32 in rs-platform-value? should everyone be using that one?

File: packages/rs-drive/src/drive/verify/contract/verify_contract/v0/mod.rs (hunk 5 above)
Comment (PastaPastaPasta) [reply packages/rs-drive/src/drive/verify/contract/verify_contract/v0/mod.rs:173]: Why?

Comment (PastaPastaPasta) [reply packages/rs-drive/src/drive/verify/contract/verify_contract/v0/mod.rs]: 2899837cca0a8214ef0ed12723cbf16e8ce85f53

File: packages/rs-drive/src/drive/verify/mod.rs (hunk 1 above)
Comment (PastaPastaPasta) [reply packages/rs-drive/src/drive/verify/mod.rs]: 2899837cca0a8214ef0ed12723cbf16e8ce85f53

Comment (PastaPastaPasta) [reply packages/rs-drive/src/drive/verify/mod.rs]: 2899837cca0a8214ef0ed12723cbf16e8ce85f53

**Review by shumkov — APPROVED** [review]
👍

**Pull request merged**